### DISCUSSION TITLE
Does the sdk of nuclei not load the specified templates.Template directly?

### DISCUSSION THREAD

**Discussion by sairson** [projectdiscovery/nuclei#4721] [Q&A] [answered]
Does the sdk of nuclei not load the specified templates.Template directly?

**Comment by sairson** [comment]
nuclei sdk does not execute directly with the parsed templates.Template structure, or do you need to use local or remote yaml to execute, and then add the parsed templates.Template to the sdk

**Comment by olearycrew** [comment]
I'm not sure I completely understand the question @sairson but you would need to parse the templates using the methods provided by the SDK and pass those to the nuclei SDK for processing. You should check out the examples here: https://github.com/projectdiscovery/nuclei/blob/dev/lib/README.md

**Comment by sairson** [comment]
I checked the relevant documents, I mean the specified data parsing as templates, Template structure, through the call, similar to the exec (target []string, templates []*templats.Template, resultCallback func()) structure

**Reply by sairson** [reply]
The current sdk seems to be able to execute templates only in the form of a specified local path, and I want to be able to parse and run from the specified template bytes

**Reply by sairson** [reply]
@olearycrew

**Comment by sairson** [comment]
That is, I want to execute the nuclei engine from a template of bytes, not a path

**Reply by olearycrew** [reply]
That's a good question - @tarunKoyalwar do you have any hints here about how that might be possible?

**Comment by tarunKoyalwar** [comment, marked as answer]
@sairson ,you can parse a template from bytes using this method from v3 sdk ( see: https://pkg.go.dev/github.com/projectdiscovery/nuclei/v3@v3.1.10/lib#NucleiEngine.ParseTemplate )

> func (*NucleiEngine) [ParseTemplate](https://github.com/projectdiscovery/nuclei/blob/v3.1.10/lib/sdk.go#L139) [¶](https://pkg.go.dev/github.com/projectdiscovery/nuclei/v3@v3.1.10/lib#NucleiEngine.ParseTemplate)
> added in v3.0.3
> func (e *[NucleiEngine](https://pkg.go.dev/github.com/projectdiscovery/nuclei/v3@v3.1.10/lib#NucleiEngine)) ParseTemplate(data [][byte](https://pkg.go.dev/builtin#byte)) (*[templates](https://pkg.go.dev/github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/templates).[Template](https://pkg.go.dev/github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/templates#Template), [error](https://pkg.go.dev/builtin#error))
> ParseTemplate parses a template from given data template verification status can be accessed from template.Verified

and to execute this obtained template you can call execute method on that type/object

```go
    tmpl ,err := nuclei.ParseTemplate(....)
    if err != nil{
       return err
    }
    tmpl.Executer.Execute(scanCtx)
```
see: https://pkg.go.dev/github.com/projectdiscovery/nuclei/v3@v3.1.10/pkg/protocols#Executer

**Reply by sairson** [reply]
thanks

**Reply by sairson** [reply]
This was a successful approach, and I appreciate it, but finally I wonder if I can specify bytes of multiple templates for scanning at the same time, or if I need to wrap the related functions
